server/api/src/routes/users.rs

```diff
@@ -0,0 +1,84 @@
+use super::common::ValidatedJson;
+use super::error::ErrorMsg;
+use super::AppState;
+use crate::auth::{self, AUTH_COOKIE_NAME};
+use axum::extract::{Path, State};
+use axum::{http::StatusCode, Json};
+use axum_extra::extract::{cookie::Cookie, PrivateCookieJar};
+use entity::user;
+use pwhash::bcrypt;
+use sea_orm::DatabaseConnection;
+use veryrezsi_core::dto::users::{LoginRequest, NewUserRequest};
+use veryrezsi_core::logic::user_operations;
+
+pub async fn login(
+    cookies: PrivateCookieJar,
+    State(ref conn): State<DatabaseConnection>,
+    ValidatedJson(login_data): ValidatedJson<LoginRequest>,
+) -> Result<PrivateCookieJar, ErrorMsg<()>> {
+    let Some(user) = user_operations::find_user_by_email(conn, login_data.email).await? else {
+        return Err(ErrorMsg::new(
+            StatusCode::UNAUTHORIZED,
+            "invalid credentials",
+        ));
+    };
+    if !user.activated {
+        return Err(ErrorMsg::new(
+            StatusCode::BAD_REQUEST,
+            "account not activated",
+        ));
+    };
+    if !bcrypt::verify(login_data.password, &user.pw_hash) {
+        return Err(ErrorMsg::new(
+            StatusCode::UNAUTHORIZED,
+            "incorrect credentials",
+        ));
+    };
+    let mut cookie = Cookie::new(auth::AUTH_COOKIE_NAME, user.id.to_string());
+    cookie.set_path("/api");
+    Ok(cookies.add(cookie))
+}
+
+pub async fn me(
+    user: auth::AuthenticatedUser,
+    State(ref conn): State<DatabaseConnection>,
+) -> Result<Json<user::Model>, ErrorMsg<()>> {
+    match user_operations::find_user_by_id(conn, user.id).await? {
+        Some(user) => Ok(Json(user)),
+        None => Err(ErrorMsg::new(StatusCode::NOT_FOUND, "user not found")),
+    }
+}
+
+pub async fn logout(cookies: PrivateCookieJar) -> Result<PrivateCookieJar, ErrorMsg<()>> {
+    match cookies.get(AUTH_COOKIE_NAME) {
+        Some(cookie) => Ok(cookies.remove(cookie)),
+        None => Err(ErrorMsg::new(StatusCode::BAD_REQUEST, "not logged in")),
```

Comment:
_Just mumbling here_ 
I was trying to find the best-practice for this. As this is a typical "Operation failed successfully" case. 
Probably better to keep it strict on the backend and handle gently on the frontend.

Reply:
Well TBH I did not think this through at all, maybe it is unnecessary to throw a bad request here, I may have a different opinion on my next refactor round and this will change then, as always. 😄

Reply:
😀